omnigibson/prims/entity_prim.py

```diff
@@ -954,76 +954,138 @@ def get_relative_linear_velocity(self):
+            if og.sim.is_stopped():
+                return XFormPrim.get_position_orientation(self, frame=frame)
+            # Delegate to RigidPrim if we are not articulated
+            elif self._articulation_view is None:
+                return self.root_link.get_position_orientation(frame=frame)
+            # Sim is running and articulation view exists, so use that physx API backend
+            else:
+                positions, orientations = self._articulation_view.get_local_poses()
 
-    def set_local_pose(self, position=None, orientation=None):
-        # If kinematic only, clear cache for the root link
-        if self.kinematic_only:
-            self.root_link.clear_kinematic_only_cache()
-        # If the simulation isn't running, we should set this prim's XForm (object-level) properties directly
-        if og.sim.is_stopped():
-            return XFormPrim.set_local_pose(self, position, orientation)
-        # Delegate to RigidPrim if we are not articulated
-        elif self._articulation_view is None:
-            self.root_link.set_local_pose(position=position, orientation=orientation)
-        # Sim is running and articulation view exists, so use that physx API backend
         else:
-            if position is not None:
-                position = np.asarray(position)[None, :]
-            if orientation is not None:
-                orientation = np.asarray(orientation)[None, [3, 0, 1, 2]]
-            self._articulation_view.set_local_poses(position, orientation)
-            PoseAPI.invalidate()
+            raise ValueError(f"Invalid frame: {frame}")
+
+        return positions[0], orientations[0][[1, 2, 3, 0]]
+
+    def set_local_pose(self, position=None, orientation=None, frame=RelativeFrame.PARENT):
+
+        import warnings
+
+        warnings.warn(
+            "set_local_pose is not implemented for articulated objects. Use set_position_orientation(frame=RelativeFrame.PARENT) instead."
+        )
+        return self.set_position_orientation(position=position, orientation=orientation, frame=RelativeFrame.PARENT)
 
     def get_local_pose(self):
-        # If the simulation isn't running, we should read from this prim's XForm (object-level) properties directly
-        if og.sim.is_stopped():
-            return XFormPrim.get_local_pose(self)
-        # Delegate to RigidPrim if we are not articulated
-        elif self._articulation_view is None:
-            return self.root_link.get_local_pose()
-        # Sim is running and articulation view exists, so use that physx API backend
-        else:
-            positions, orientations = self._articulation_view.get_local_poses()
-            return positions[0], orientations[0][[1, 2, 3, 0]]
+
+        import warnings
+
+        warnings.warn(
+            "get_local_pose is not implemented for articulated objects. Use get_position_orientation(frame=RelativeFrame.PARENT) instead."
+        )
+        return self.get_position_orientation(frame=RelativeFrame.PARENT)
```

Comment:
Replace all messages to say "get_local_pose is deprecated and will be removed in a future release. Use get_position_orientation(frame=RelativeFrame.PARENT) instead"